### DISCUSSION TITLE
how does take work in include option?

### DISCUSSION THREAD

**Discussion by Knorway** [prisma/prisma#15591] [Q&A] [answered]
I have these 3 models. user, post, comment and a user has many posts and post has many comments.

```
model user {
  id        BigInt     @id @default(autoincrement())
  comment   comment[]
  post      post[]
}

model post {
  id         BigInt    @id @default(autoincrement())
  user_id    BigInt?
  user       user?     @relation(fields: [user_id], references: [id])
  comment    comment[]
}

model comment {
  id         BigInt    @id @default(autoincrement())
  post_id    BigInt?
  user_id    BigInt?
  user       user?     @relation(fields: [user_id], references: [id])
  post       post?     @relation(fields: [post_id], references: [id])
}
```

what I'm trying to achieve is to get the posts of a certain user with limited comment. i've tried query below and succeeded.
```
const posts = await prisma.post.findMany({
		where: {
			{ user_id: 9 },
		},
		include: {
			comment: { take: 3 },
		},
		orderBy: { id: 'desc' },
	});
```

but generated query seems bit suspicious
```
prisma:query SELECT `jstagram_dev`.`post`.`id`, `jstagram_dev`.`post`.`content`, `jstagram_dev`.`post`.`img_url`, `jstagram_dev`.`post`.`user_id`, `jstagram_dev`.`post`.`created_at`, `jstagram_dev`.`post`.`updated_at` FROM `jstagram_dev`.`post` WHERE `jstagram_dev`.`post`.`user_id` = ? ORDER BY `jstagram_dev`.`post`.`id` DESC

// comments part
prisma:query SELECT `jstagram_dev`.`comment`.`id`, `jstagram_dev`.`comment`.`content`, `jstagram_dev`.`comment`.`post_id`, `jstagram_dev`.`comment`.`user_id`, `jstagram_dev`.`comment`.`created_at`, `jstagram_dev`.`comment`.`updated_at` FROM `jstagram_dev`.`comment` WHERE `jstagram_dev`.`comment`.`post_id` IN (?,?,?,?,?,?) ORDER BY `jstagram_dev`.`comment`.`id` ASC
```
it seems like prisma fetches all the comments that's related to the posts and makes chunk in memory and allocating them, not limiting in the first place. i guess it's completely reasonable because it's not an easy job for any orms but it could be quite risky if there's thousands of comments in db. i just wanted to know if it's true and if it is, i'd like to get some advice how to get limited comments on each post in prisma avoiding n + 1 with prisma. thank you!

**Comment by austincrim** [comment, marked as answer]
Hey @Knorway 👋,

If you see noticeable performance issues with this query, I would definitely open a [bug report](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) with more details.

You can also rewrite it using [$queryRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw) if you need fine-grained control.

**Comment by Knorway** [comment]
@austincrim I haven't encountered performance issue with this one yet. my question is about is prisma actually limiting the result under the hood or not. because generated query seems bit different from one from [this video](https://www.youtube.com/watch?v=7oMfBGEdwsc&t=1230s) and I think it could be a reason why he's so struggling with performance in the video. he's using `take` 3 depth and yes I already ended up using $queryRaw for ROW_NUMBER()

**Reply by austincrim** [reply]
I assume the generated query you posted above is what Prisma uses. I don't think it does any limiting under-the-hood.

**Reply by Knorway** [reply]
> I assume the generated query you posted above is what Prisma uses. I don't think it does any limiting under-the-hood.

I got it. Thank you for your answer!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
